sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -788,14 +789,14 @@ def get_stage_key(stage):
   pcoll_id_remap = {}
   remaining_stages = []
   for sibling_stages in grouped_eligible_stages.values():
+    if len(sibling_stages) == 1:
+      ineligible_stages.extend(sibling_stages)
```

Comment:
dumb question: what's a key-with-none stage? Is it obvious why a single-sibling stage should be marked 'ineligible' or it needs a comment? Thanks.

Reply:
KeyWithNone / KeyWithVoid refers to adding None as a key in the expansion for CombineGlobally [here](https://github.com/apache/beam/blob/5291b2c88e4371d7e9eb1ac6708dd4dc3bf17f8e/sdks/python/apache_beam/transforms/core.py#L2006-L2008). Basically this optimizer is finding sibling duplicate KeyWithNone stages with the same input and eliminating them.

I'm reverting this block because it's not necessary after the other changes - it was an alternate workaround that I was trying that would also allow the newly added tests to pass.